File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        num_count = {}

        for i in nums:
            if i in num_count:
                num_count[i] += 1
                if num_count[i] == 2: return True
            else:
                num_count[i] = 1
        
        return False

File: Data Structures & Algorithms/sort-colors/submission-0.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        // Assuming array only contains 0, 1, or 2
        int counts[] = {0, 0 ,0};

        // Count the quantity of each val in nums
        for (int n: nums) {
            counts[n]++;
        }

        int i = 0;
        for (int n = 0; n < 3; n++) {
            for (int j = 0; j < counts[n]; j++) {
                nums[i] = n;
                i++;
            }
        }
    }
};